src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -62,6 +81,43 @@ void check_sshfs_exists(mp::SSHSession& session)
     run_cmd(session, "which sshfs", error_handler);
 }
 
+// If the target name starts with ~, this function returns the full directory name. If not, it returns the same
+// target name.
+std::string expand_home_directory(mp::SSHSession& session, const std::string& target)
+{
+    std::string expanded_target;
+
+    if ('~' == target[0])
+    {
+        std::string home;
+
+        // Get the user name, which can be empty.
+        auto pos = (1 == target.size()) ? 1 : target.find('/', 1);
+        if (1 == pos)
+        {
+            home = run_string_cmd(session, "pwd");
+        }
```

Comment:
I would rather execute `getent passwd $USER...` for this case.

Reply:
Right.

Reply:
Why not `echo ~`? It expands to ${HOME}, which is by default the value given by `getent`. Should the user change ${HOME}, I guess he would expect ~ to expand to the new value.

Reply:
If we go for bash expanding it, we don't need the "~" or "~user" check at all.

Reply:
Right, it also simplifies the code